### ISSUE TITLE
Compile failure on FPC trunk.

### ISSUE THREAD

**Issue by LongDirtyAnimAlf** [bgrabitmap/bgrabitmap#185]
Hi,

When using your wonderfull BGRABitMap in a runtime creating mode (no install of packages), it will not compile on FPC trunk.
FPC trunk changed the implementation of the \*-operator for TPointF. In trunk, this operator now returns a TPointF also.
FPC trunk added a  \*\*-operator for TPointF. In trunk, this operator now returns a single.
Proposal. Add \*\* operator into the record helpers for TPointF inside geometrytypes.inc, and change all TPointF\*TPointF into TPointF\*\*TPointF.
{$if FPC_FULLVERSION<030301}
operator \*\* (const pt1, pt2: TPointF): single; inline;
{$endif}
This approach has been tested on FPC trunk by me, and works well.

Thanks !

**Comment by circular17** [comment]
Hi !

Thanks for your message. Is there a discussion somewhere about the change of implementation of this operator? I would like to ask not to change it, this breaks existing code.

Regards

**Comment by Alexey-T** [comment]
FPC truck patch- https://gitlab.com/freepascal.org/fpc/source/-/commit/f1e391fb415239d926c4f23babe812e67824ef95

FPC issue- https://gitlab.com/freepascal.org/fpc/source/-/issues/40042

Pls change all `*` product to `**` product. But problem is not ended here. I changed locally. And now I face another error!
```
procedure TBGRAVectorizedFont.WriteCustomHeader(AStream: TStream);
var metric: TFontPixelMetric;
begin
  inherited WriteCustomHeader(AStream);
  LEWriteLongint(AStream, length(FName));
  AStream.Write(FName[1],length(FName));
  LEWriteLongint(AStream, integer(FStyle));   /// Here - cannot convert TFontStyles to LongInt
```

**Comment by circular17** [comment]
Thanks for the links.

I don't like it, but well, that's more Delphi compatible I suppose.

Indeed changing the `*` for `**` would solve that.

Regarding the case from TFontStyles to LongInt, is there another way of converting that, apart from explicitly computing the bits of the integer?

**Comment by Alexey-T** [comment]
TFontStyles. Seems you need the custom func, like this
```
function Canvas_NumberToFontStyles(Num: integer): TFontStyles;
begin
  Result:= [];
  if (Num and FONT_B)<>0 then Include(Result, fsBold);
  if (Num and FONT_I)<>0 then Include(Result, fsItalic);
  if (Num and FONT_U)<>0 then Include(Result, fsUnderline);
  if (Num and FONT_S)<>0 then Include(Result, fsStrikeOut);
end;
```
FONT_x are my bit flags.

**Comment by circular17** [comment]
Hmm yeah.

Thank you for your point of view and code sample.

**Comment by Alexey-T** [comment]
Try the last trunk again, they changed the SizeOf(TFontStyles) 4 days ago

**Comment by circular17** [comment]
I applied the changes on dev branch.

Does it fix the problem for you?

**Comment by Alexey-T** [comment]
Yes, thanks

**Comment by regs01** [comment]
Dev branch has a compilation error

`TBGRADefaultBitmap.InternalTextOutCurved` has a variable `angle` and couple of lines
```pascal
    with ACursor.CurrentTangent do angle := arctan2(y,x);
    with ACursor.CurrentCoordinate do
    begin
      if ATexture = nil then
        TextOutAngle(x,y, system.round(-angle*1800/Pi), currentGlyphUtf8, AColor, taCenter)
      else
        TextOutAngle(x,y, system.round(-angle*1800/Pi), currentGlyphUtf8, ATexture, taCenter);
    end;  
```
`CurrentTangent` and `CurrentCoordinate` are `TPointF` and `TPointF` has a function `Angle`

So renaming angle variable to iAngle fixes it

**Comment by Alexey-T** [comment]
@LongDirtyAnimAlf also pls fold constants here: `-angle*1800/Pi` -> `-angle*(1800/Pi)`

**Comment by circular17** [comment]
fix on dev branch

**Comment by lainz** [comment]
Thank you

**Comment by Shebuka** [comment]
Hello @circular17 , when will you make a new release with this fix?

**Comment by circular17** [comment]
Here i've done it for you

[BGRABitmap v11.5.4](https://github.com/bgrabitmap/bgrabitmap/releases/tag/v11.5.4)

**Comment by Shebuka** [comment]
That's awesome! Thanks!
Can you tag bgracontrols too?

**Comment by circular17** [comment]
I've published BGRAControls too (v7.7).

I don't know anymore where the thread is to announce the version.
